cli/src/main/kotlin/commands/ScannerCommand.kt

```diff
@@ -236,20 +243,31 @@ class ScannerCommand : CliktCommand(name = "scan", help = "Run external license
         }
 
         // Configure the package and project scanners.
-        val scanner = scannerFactory.create(config.scanner, config.downloader)
-        val projectScanner = projectScannerFactory?.create(config.scanner, config.downloader) ?: scanner
+        val defaultScanner = scannerFactory.create(config.scanner, config.downloader)
+        val packageScanner = defaultScanner.takeIf { DataEntity.PACKAGES in entities }
+        val projectScanner = (projectScannerFactory?.create(config.scanner, config.downloader) ?: defaultScanner)
+            .takeIf { DataEntity.PROJECTS in entities }
+
+        if (projectScanner != packageScanner) {
+            if (projectScanner != null) {
+                println("Using project scanner '${projectScanner.scannerName}'.")
+            } else {
+                println("Projects will not be scanned.")
+            }
 
-        if (projectScanner != scanner) {
-            println("Using project scanner '${projectScanner.scannerName}'.")
-            println("Using package scanner '${scanner.scannerName}'.")
+            if (packageScanner != null) {
+                println("Using package scanner '${packageScanner.scannerName}'.")
+            } else {
+                println("Packages will not be scanned.")
+            }
         } else {
-            println("Using scanner '${scanner.scannerName}'.")
+            println("Using scanner '${defaultScanner.scannerName}'.")
```

Comment:
What if `entities` is empty (provided that is even possible with the option)?

Reply:
> provided that is even possible with the option

It's not possible: The option defaults to all entities, and if the option *is* specified, it also must have a value and cannot be empty.